walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,84 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Iterator;
 
 public final class FindMeetingQuery {
+
+  /** Returns time ranges that satisfy the request and take into account, the events */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    long duration = request.getDuration();
+
+    // when the requested meeting duration is greater than a day, there are means no viable ranges
```

Comment:
Here and other comments, please use capital letters to start and end with a period.

```suggestion
    // When the requested meeting duration is greater than a day, that means there are no viable ranges
```